### PR TITLE
feat(es/decorators): Implement decorator version `2023-11`

### LINKED ISSUE #8922: Various transform bugs with the JavaScript decorators proposal
### Describe the bug

I'm writing tests for the JavaScript decorators proposal as part of esbuild's implementation: https://github.com/evanw/decorator-tests. This issue is mainly to let you know about the tests, which you may find useful. There are over 200 test failures so I'm not going to list them all out here. In certain cases, SWC generates code with syntax errors when transforming code containing JavaScript decorators. These syntax error bugs are reported below. There are also many behavioral bugs that are not reported here but that can be found in the test suite.

### Input code

```typescript
// SWC turns this into code with a syntax error
class Foo {
    accessor foo;
    static accessor foo;
}

// SWC turns this into code with a syntax error
@dec class Bar extends (await Bar) {
    @dec static field;
}

// SWC fails to transform this decorator at all
class Baz {
    accessor foo = @dec class {}
}
```

### Config

```json
{
  "jsc": {
    "target": "es2022",
    "parser": {
      "syntax": "typescript",
      "decorators": true
    },
    "transform": {
      "decoratorVersion": "2022-03"
    }
  }
}
```

### Playground link (or link to the minimal reproduction)

https://play.swc.rs/?version=1.5.4&code=H4sIAAAAAAAAA5WPTQrCQAxG93OKb6mrHkAEUfACLlyH6ZQOjBNIIq2W3t2xrVDEjVl%2B%2BXkvVYXL9QS7S1ZYGxUxG8NzHdBFa0HQRzbqEURYnE%2BkijMzBodS5H1QZUHDvJsSNbLovxqjc9WfpEMdPGbckQSht5BrxYY6ivaOtovCNLhQmxhSvcY1FFPBMUwoa8Nym9llh4Ws%2BJGBUnIf0vPHY9hjZTOM5f4LT6dv4DgBAAA%3D&config=H4sIAAAAAAAAA02OsQrDMAxE93yF0ZxCcLd8SHfhqiGF2uakQkPIv1duQuggkO7eHVq7EOipicaw%2BuqHMSYxv0k0DjFSv%2BuVoYKTc0WXbPxppC1VNGGudtDu3iUVsBWoE4a3%2FIztaDNw1kfB67%2FwjNwEOpfcqtsLl%2BFKe7prs30BOXCfLbUAAAA%3D

### SWC Info output

_No response_

### Expected behavior

When given valid JavaScript code as input, SWC should never generate code containing syntax errors.

### Actual behavior

_No response_

### Version

1.5.4 (the latest on the playground)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kdy1**
**Description:**

https://github.com/babel/babel/blob/440fe413330f19fdb2c5fa63ffab87e67383d12d/packages/babel-helper-create-class-features-plugin/src/decorators.ts

**Related issue:**

 - Closes https://github.com/swc-project/swc/issues/9078
 - Closes https://github.com/swc-project/swc/issues/9079
 - Closes https://github.com/swc-project/swc/issues/8922

### CODE REVIEW

File: crates/swc_ecma_transforms_proposal/src/lib.rs [hunk 1]
```diff
@@ -1,11 +1,27 @@
...
+use serde::{Deserialize, Serialize};
+
 pub use self::{
     decorators::decorators, export_default_from::export_default_from,
     import_assertions::import_assertions,
 };
 
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub enum DecoratorVersion {
```
Comment (magic-akari) [inline crates/swc_ecma_transforms_proposal/src/lib.rs:13]: Babel 8 keeps only the newest and the oldest versions; I think we should also keep just two versions.

Keeping the latest version is sensible, but another that leverages TypeScript's legacy decorators (experimentalDecorators) is preferable.

It is necessary to specify the exact version. SWC references the TypeScript version with the date `2021-12`, whereas Babel uses `2018-09`.

The TypeScript documentation states that it adopts a Stage 2 version, but it doesn't specify the exact year and month of the draft.

Comment (kdy1) [reply crates/swc_ecma_transforms_proposal/src/lib.rs:13]: But some users will complain that removal of implementations is a breaking change, and I'm not sure if it's fine to remove an option.

Comment (chenjiahan) [reply crates/swc_ecma_transforms_proposal/src/lib.rs:13]: Please only remove an option in a major version 😂

**Comment by kdy1** [comment]
Closing in favor of https://github.com/swc-project/swc/pull/9488

**Pull request closed without merging**